injection/sharedmain/main.go

```diff
@@ -324,6 +330,18 @@ func MainWithConfig(ctx context.Context, component string, cfg *rest.Config, cto
 	}
 }
 
+type healthProbesDisabledKey struct{}
+
+// WithHealthProbesDisabled signals to MainWithContext that it should disable default probes (readiness and liveness).
+func WithHealthProbesDisabled(ctx context.Context) context.Context {
+	return context.WithValue(ctx, healthProbesDisabledKey{}, struct{}{})
+}
+
+// HealthProbesDisabled checks if default health checks are disabled in the related context.
+func HealthProbesDisabled(ctx context.Context) bool {
```

Comment:
Does this need to be an exported method, or can it be internal-only?

Reply:
Should be private.